src/monitoring/observers/node-health.observer.ts

```diff
@@ -0,0 +1,314 @@
+import { cloneDeep } from 'lodash';
+import { BehaviorSubject } from 'rxjs';
+import { Injectable } from '@nestjs/common';
+import { Interval } from '@nestjs/schedule';
+import { NodeClient, NodeMode } from 'src/ain/node/node-client';
+import { NodeError, NodeService, NodeType } from 'src/ain/node/node.service';
+import { MetricObserver } from 'src/monitoring/metric.observer';
+import { MonitoringService } from 'src/monitoring/monitoring.service';
+import { MailService } from 'src/shared/services/mail.service';
+
+interface NodesData {
+  connectedNodes: Map<NodeType, BehaviorSubject<NodeClient | null>>;
+  configuredNodes: Map<NodeType, Record<NodeMode, NodeClient | null>>;
+  errors: NodeError[];
+}
+
+interface NodesHealth {
+  allNodesHealthy: boolean;
+  nodes: {
+    [key in NodeType]: NodePairHealth;
+  };
+}
+
+type NodePairHealth = {
+  mails: NodePairHealthMails;
+  health: {
+    [key in NodeMode]: NodeHealth;
+  };
+};
```

Comment:
This looks a bit weird to me ... let's discuss at the daily.

Reply:
sure, lets discuss